### PR TITLE
Add Support for XSS Protection, Content Type Options, Content Security Policy, Referrer Policy, and Strict Transport Security

### PR DESCRIPTION
**Pull request by eric-gonzales**
This PR adds support for: 

- X-XSS-Protection headers
- X-Content-Type-Options headers
- Content-Security-Policy headers
- Referrer-Policy headers
- Strict-Transport-Security headers

### CODE REVIEW

**Comment by dbpfindexp** [comment]
Anyway we can PR into the main plugin and have the plugin owner incorporate these or is this specific to us?

**Comment by dbpfindexp** [comment]
Just wondering if we can keep the management of the plugin off our repositories since i assume this will continue to chagne.

**Comment by eric-gonzales** [comment]
@dbpfindexp - will definitely make a PR to the original owner's repo in addition to this one. Was thinking our need for this functionality is urgent at the moment, which is why I forked it instead so we can get it out soon. Not to mention we have reasons to maintain both a Grails 3.x and a Grails 2.x version of this plugin.

File: src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/ReferrerPolicyConfiguration.groovy [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    final static boolean DEFAULT_ENABLED = true
+    final static boolean DEFAULT_NO_REFERRER = false
+    final static boolean DEFAULT_NO_REFERRER_WHEN_DOWNGRADE = true
+    final static boolean DEFAULT_ORIGIN = false
+    final static boolean DEFAULT_ORIGIN_WHEN_CROSS_ORIGIN = false
+    final static boolean DEFAULT_SAME_ORIGIN = false
+    final static boolean DEFAULT_STRICT_ORIGIN = false
+    final static boolean DEFAULT_STRICT_ORIGIN_WHEN_CROSS_ORIGIN = false
+    final static boolean DEFAULT_UNSAFE_URL = false
+    final static String DEFAULT_URL_PATTERN = '/*'
```
Comment (Joshua-Brent-Collins) [inline src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/ReferrerPolicyConfiguration.groovy]: Have we tested these default url patterns also captures the root (no /) hostname ?

File: src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/XXSSProtectionConfiguration.groovy [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        if (disabled) {
+            return DISABLED
+        }
+
+        if (enabled) {
+            return ENABLED
+        }
+
+        return ENABLED_BLOCK
```
Comment (Joshua-Brent-Collins) [inline src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/XXSSProtectionConfiguration.groovy]: Might want to glance over these sections and make sure the naming conventions are all correct/accessible.

**Comment by dbpfindexp** [comment]
@eric-gonzales  - agreed we should pub this to our custom repo to become compliant but we should def put a big TODO to essentially get this approved by the OG creator so we can just leverage that going forward and offload maintaining this long term.

File: src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/ReferrerPolicyConfiguration.groovy (hunk 1 above)
Comment (eric-gonzales) [reply src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/ReferrerPolicyConfiguration.groovy]: Yes, works locally

**Comment by eric-gonzales** [comment]
@Joshua-Brent-Collins - added tests and addressed comments!

**Pull request merged**